src/StringTools/StringTools.csproj

```diff
@@ -0,0 +1,38 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>$(LibraryTargetFrameworks)</TargetFrameworks>
+    <TargetFrameworks Condition="'$(MSBuildRuntimeType)' != 'Core' and '$(MonoBuild)' != 'true'">$(LibraryTargetFrameworks);net35</TargetFrameworks>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <WarningsAsErrors>true</WarningsAsErrors>
+    <IsPackable>true</IsPackable>
+    <LangVersion>8.0</LangVersion>
+    <PackageId>Microsoft.NET.StringTools</PackageId>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+
+    <VersionPrefix>1.0.0</VersionPrefix>
+    <AssemblyVersion>1.0.0.0</AssemblyVersion>
+    <PreReleaseVersionLabel>preview</PreReleaseVersionLabel>
+
+    <AssemblyName>StringTools</AssemblyName>
+    <AssemblyName Condition="'$(TargetFramework)' == 'net35'">StringTools.net35</AssemblyName>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' != 'net35'">
```

Comment:
The previous string interning implementation went to great lengths to also work in 3.5 so I'm afraid that not doing anything at all could lead to impactful regressions. Although I agree that perf regressions may be acceptable, especially when it comes to memory _in one specific process_.

Has the importance of the 3.5 task host gone down compared to, say, five years ago, or is it still about the same?

It's code churn either way - moving the existing 3.5 weak interner into the library or stubbing it out. And since the code already exists and the library may potentially find 3.5 uses outside of MSBuild, I'm leaning towards leaving it in for now.